vernac/g_vernac.mlg

```diff
@@ -494,6 +494,10 @@ let warn_deprecated_include_type =
   CWarnings.create ~name:"deprecated-include-type" ~category:"deprecated"
          (fun () -> strbrk "Include Type is deprecated; use Include instead")
 
+let warn_deprecated_as_ident_kind =
+  CWarnings.create ~name:"deprecated-as-ident-kind" ~category:"deprecated"
+         (fun () -> strbrk "grammar kind \"as ident\" was formerly accepting any identifier or \"_\"; it now accepts only identifiers; use \"as name\" if you expect \"_\" to be parsed too, or silence the warning if you actually intended to accept only identifiers.")
```

Comment:
```suggestion
         (fun () -> strbrk "grammar kind \"as ident\" no longer accepts \"_\"; use \"as name\" instead to accept \"_\", too, or silence the warning if you actually intended to accept only identifiers.")
```

How difficult would it be to show this message (with different text) only when `_` is used for `ident`?  That would avoid the false positives, easier for users.

Reply:
Indeed less convoluted!

Reply:
> How difficult would it be to show this message (with different text) only when _ is used for ident? 

It would not be difficult but of no real use. The update has to be done at the time of defining the `Notation`, not at the time of using it.